cocos2d/core/sprites/CCSprite.js

```diff
@@ -645,14 +634,21 @@ cc.Sprite = cc.Node.extend(/** @lends cc.Sprite# */{
         var tex = cc.textureCache.getTextureForKey(filename);
         if (!tex) {
             tex = cc.textureCache.addImage(filename);
-            return this.initWithTexture(tex, rect || cc.rect(0, 0, tex._contentSize.width, tex._contentSize.height));
-        } else {
-            if (!rect) {
-                var size = tex.getContentSize();
-                rect = cc.rect(0, 0, size.width, size.height);
-            }
-            return this.initWithTexture(tex, rect);
         }
+        this._loader.clear();
```

Comment:
You can put it into the if section `if (!tex.isLoaded())`, because `initWithTexture` will clear the loader by itself